core/src/main/java/fr/sncf/osrd/railjson/parser/RJSStandaloneTrainScheduleParser.java

```diff
@@ -57,9 +58,20 @@ public static StandaloneTrainSchedule parse(
                 allowances.add(parseAllowance(envelopeSimPath, rjsAllowance));
             }
 
+        // parse timed waypoints
+        var scheduledPoints = new ArrayList<ScheduledPoint>();
+        if (rjsTrainSchedule.scheduledPoints != null) {
+            for (var rjsSchedulePoints : rjsTrainSchedule.scheduledPoints) {
+                if (rjsSchedulePoints.position < 0)
+                    scheduledPoints.add(new ScheduledPoint(trainPath.length(), rjsSchedulePoints.time));
+                else
+                    scheduledPoints.add(new ScheduledPoint(rjsSchedulePoints.position, rjsSchedulePoints.time));
```

Comment:
I'd be nice to somewhat validate that the point is within the bounds of the path when positive, or at least clamp it back within range

Reply:
Done! If the position is greater than the path length I throw an error.